[Bug]: panic: runtime error: invalid memory address or nil pointer dereference after manually deleting server

### What happened?

We deleted a server from outside Terraform, and I'm guessing it broke because the rdns entry was also suddenly missing. Running `terraform plan` crashes.

I ran `$ terraform state rm 'module.platform.hcloud_rdns.masters["k8s-master3-non-prod"]'` and that repaired the situation.

```
Terraform v1.0.5
on linux_amd64
Configuring remote state backend...
Initializing Terraform configuration...
module.platform.random_string.certificate_key: Refreshing state... [id=CEcl5Xz2PIUu]
module.platform.random_string.bootstrap_token_id: Refreshing state... [id=aeq8yb]
module.platform.random_string.bootstrap_token_secret: Refreshing state... [id=zdho6cy69wm661cf]
module.platform.hcloud_network.private: Refreshing state... [id=1201870]
module.platform.hcloud_load_balancer.inbalancer: Refreshing state... [id=414004]
module.platform.hcloud_firewall.node_firewall: Refreshing state... [id=125994]
module.platform.hcloud_ssh_key.default["1"]: Refreshing state... [id=4309276]
module.platform.hcloud_ssh_key.default["0"]: Refreshing state... [id=4309278]
module.platform.hcloud_ssh_key.default["2"]: Refreshing state... [id=4309277]
module.platform.hcloud_placement_group.k8s-nodes: Refreshing state... [id=671]
module.platform.hcloud_network_subnet.k8s-nodes: Refreshing state... [id=1201870-10.67.1.0/24]
module.platform.hcloud_server.masters[1]: Refreshing state... [id=14043501]
module.platform.hcloud_server.masters[2]: Refreshing state... [id=14043308]
module.platform.hcloud_server.masters[0]: Refreshing state... [id=14043182]
module.platform.hcloud_load_balancer_network.inbalancernetwork: Refreshing state... [id=414004-1201870]
module.platform.hcloud_rdns.inbalancerrevdns: Refreshing state... [id=l-414004-167.233.12.201]
module.platform.hcloud_load_balancer_service.inbalancer_https: Refreshing state... [id=414004__443]
module.platform.hcloud_load_balancer_service.inbalancer_http: Refreshing state... [id=414004__80]
module.platform.hcloud_load_balancer_target.load_balancer_target: Refreshing state... [id=lb-label-selector-tgt-86b01528117ce2703b09ffc2adeb338afe78b944850868f246c7cf222db0c369-414004]
module.platform.hcloud_rdns.masters["k8s-master3-non-prod"]: Refreshing state... [id=s-14043308-149.12.193.58]
module.platform.hcloud_rdns.masters["k8s-master1-non-prod"]: Refreshing state... [id=s-14043182-95.201.127.46]
module.platform.hcloud_rdns.masters["k8s-master2-non-prod"]: Refreshing state... [id=s-14043501-123.88.57.105]
╷
│ Error: Plugin did not respond
│
│   with module.platform.hcloud_rdns.masters["k8s-master1-non-prod"],
│   on .terraform/modules/platform/main.tf line 110, in resource "hcloud_rdns" "masters":
│  110: resource "hcloud_rdns" "masters" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.platform.hcloud_rdns.masters["k8s-master2-non-prod"],
│   on .terraform/modules/platform/main.tf line 110, in resource "hcloud_rdns" "masters":
│  110: resource "hcloud_rdns" "masters" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.platform.hcloud_rdns.masters["k8s-master3-non-prod"],
│   on .terraform/modules/platform/main.tf line 110, in resource "hcloud_rdns" "masters":
│  110: resource "hcloud_rdns" "masters" {
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-hcloud_v1.30.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xa57e8c]

goroutine 212 [running]:
github.com/hetznercloud/hcloud-go/hcloud.(*Server).GetDNSPtrForIP(0x0, {0xc0003a82a0, 0x10, 0x10})
	github.com/hetznercloud/hcloud-go@v1.31.1/hcloud/server.go:173 +0x2c
github.com/hetznercloud/terraform-provider-hcloud/internal/rdns.resourceReverseDNSRead({0x11103b8, 0xc0005fb8c0}, 0xc0000a5e00, {0xe4d840, 0xc0004586c0})
	github.com/hetznercloud/terraform-provider-hcloud/internal/rdns/resource_hcloud_rdns.go:71 +0x174
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc000209500, {0x1110348, 0xc0008e4c00}, 0x24, {0xe4d840, 0xc0004586c0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:347 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000209500, {0x1110348, 0xc0008e4c00}, 0xc00031b6c0, {0xe4d840, 0xc0004586c0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:624 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0000ac000, {0x1110348, 0xc0008e4c00}, 0xc0008e4c40)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:575 +0x4bd
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc0000a0860, {0x11103f0, 0xc0005d8570}, 0xc0005fb4a0)
	github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/server/server.go:298 +0x1fb
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xed2b20, 0xc0000a0860}, {0x11103f0, 0xc0005d8570}, 0xc0005fb440, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000c4700, {0x1121310, 0xc000156780}, 0xc0006207e0, 0xc0005c2300, 0x1758e90, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0000c4700, {0x1121310, 0xc000156780}, 0xc0006207e0, 0x0)
	google.golang.org/grpc@v1.40.0/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.40.0/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.40.0/server.go:939 +0x294

Error: The terraform-provider-hcloud_v1.30.0 plugin crashed!
```

### What did you expect to happen?

No crash, just a plan.

### Please provide a minimal working example

n/a

--- a/internal/rdns/resource_rdns.go
+++ b/internal/rdns/resource_rdns.go
@@ -3,6 +3,7 @@ package rdns
 import (
 	"context"
 	"fmt"
+	"log"
 	"net"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
@@ -107,18 +108,27 @@ func resourceReverseDNSCreate(ctx context.Context, d *schema.ResourceData, m int
 
 	var rdns hcloud.RDNSSupporter
 	var err error
+	var resourceName string
 
 	switch {
 	case serverOK:
 		rdns, _, err = c.Server.GetByID(ctx, serverID.(int))
+		resourceName = "Server"
 	case floatingIPOK:
 		rdns, _, err = c.FloatingIP.GetByID(ctx, floatingIPID.(int))
+		resourceName = "Floating IP"
 	case loadBalancerOK:
 		rdns, _, err = c.LoadBalancer.GetByID(ctx, loadBalancerID.(int))
+		resourceName = "Load Balancer"
 	}
-
 	if err != nil {
 		return hcclient.ErrorToDiag(err)
+	}
+
+	if rdns == nil {
+		log.Printf("[WARN] %s (%s) not found, removing from state", resourceName, d.Id())
+		d.SetId("")
+		return nil
 	}
 
 	d.SetId(generateRDNSID(rdns, ip))
@@ -246,6 +256,9 @@ func lookupRDNSID(ctx context.Context, terraformID string, client *hcloud.Client
 		rdns, _, err = client.LoadBalancer.GetByID(ctx, id)
 	default:
 		err = InvalidRDNSIDError{terraformID}
+	}
+	if rdns == nil {
+		return nil, nil, InvalidRDNSIDError{terraformID}
 	}
 
 	return rdns, ip, err

